### ISSUE TITLE
Installations section is a bit confusing 

### ISSUE THREAD

**Issue by qparis** [PhoenicisOrg/phoenicis#1599]
When I install a new program on Phoencis, I'm always disturbed by the behavior of the "installations" pane. 

 * The distinction between "installations" and "containers" is not clear (wording issue) 
 * It is a bit unexpected to use a "galery" widget to navigate between wizards. (Don't know if my thoughts are clear or not). 

Idea: I would to move this section on the right of the window with an icon instead of "Installations". To emphasize the fact that something is happening right now for the user, I would also add a badge corresponding to the number of currently opened wizards 

<img width="1271" alt="capture d ecran 2018-12-18 a 21 55 48" src="https://user-images.githubusercontent.com/373645/50182572-f6b59980-030f-11e9-9648-a93802eb175d.png">

If you have any other idea

**Comment by t0msk** [comment]
I agree, something like Tasks / Notification icon.

Or make that Installations panel will show only when something is installing?

**Comment by madoar** [comment]
So basically you would only change the way to access the installations tab, but not its content?

**Comment by plata** [comment]
Right now, the installations tab is just a tab like everything else. I'm not sure if it is possible in JavaFX to move only one tab to the right (I think I even tried that originally).

**Comment by plata** [comment]
This might be an option: https://stackoverflow.com/questions/47465267/add-a-button-to-the-right-of-tabpane

**Comment by madoar** [comment]
Most ideal would be if we could select the last tab via css (`:last-child`), but this doesn't seem to be possible.

If we think about adding a tab outside the scope of `TabPane`, why don't we think bigger and talk about fully replacing the `TabPane` by a custom component?
This component could use a `BorderPane` for its main layout. The top section of the `BorderPane` could then be a `HBox`, which can be used to arrange the tab buttons at the top. The center of the `BorderPane` can be used to show the content of the selected tab.

### Advantages:
- we have full control about what is shown, how it is shown and where it is shown

### Disadvantages:
- we basically need to write 90% of the functionality which is offered by `TabPane`

**Comment by plata** [comment]
I don't like the buttons solution because the behavior we want to achieve (i.e. panes tied to a button) is very much a `TabPane`. If we reimplement this with buttons, it's quite likely that we mess something up which would otherwise be working out of the box.

**Comment by madoar** [comment]
What are the alternatives?
I see three choices:
1) we show the installations inside the main window
2) we open a popup for each installation
3) we do the installations in the background and show nothing

Option three is bad in my opinion, because the user is shown no information and therefore can't react.
Option two leads to window cluttering, if you open a lot of installations at the same time.

Additionally, I think the word "installation" is a bit misleading.
I believe the words "environment setup" are much more fitting, because we're not installing an application but preparing an environment in which an application can then be installed.
When viewed under this aspect I think option one is the best fitting option, because the environment that is setup is an environment defined by Phoenicis and therefore belongs inside the application.

**Comment by plata** [comment]
I also think that option 1 is the best. In my opinion "installation" is the correct term because this is also used for engines and verb installations. For me it's a bit similar to the "Downloads" in Steam.

**Comment by qparis** [comment]
Installation also means "installation on hard drive". Therefore, I find it confusing. 

These windows are "wizard", they can be used for basically anything: installation of a program, uninstallation, wine downloading, verb installation, etc... 

> Option two leads to window cluttering, if you open a lot of installations at the same time.

Yup, but option one is better when you have only one or two windows opened. 
How much people will have windows open at the same time in average? I think that in 95% of the cases, user only open one window, terminate the wizard and close it but we may want to collect data to measure this. 

Option three is impossible for me. We definetely need interactions

**Comment by plata** [comment]
When I did the installation tab originally, I was asking myself: How do other applications handle this use case (wizard vs tab)? Like I said, Steam downloads was something that seemed to match (ok, they also have wizards to start the installation but they are very short).
I agree that people normally will not run multiple wizards in parallel.
I found it quite nice that the installation does not open a separate window if you install e.g. a game which downloads for hours in the background.

In the end, I don't really know what's the best solution either.

**Comment by qparis** [comment]
Yeap, this is perfectly comprehensible when it is just about downloads because you can give a progress feddback directly and at the same place. 
In our case, we are displaying the wizard somewhere else (i.e. installation tab), so it's a bit different

**Comment by plata** [comment]
If the wizard was finished after 5mins, I would definitely use a separate window. I'm not so sure it's a good idea to have a separate window which must be open for hours. This doesn't feel like a wizard for me anymore, rather like a download which needs interaction every now and then.
We could think about the Steam approach as well: Use wizards for short interactions and move long tasks to the tab. However, that might become pretty confusing if short and long tasks switch often during an installation.

**Comment by qparis** [comment]
Maybe we need to enumerate usecase and try to get an idea about how they are used

**Comment by plata** [comment]
Ok, so what do we have basically:
- engine installation
- Verb installation
- app installation

**Comment by qparis** [comment]
We could also have potentially: 

- app removal if we follow POLv4 pattern (i.e. use a wizard instead of a simple confirm window)
- shortcut creation inside a prefix (I find that POLv4 process to create a shortcut is better than POLv5 currently)
- Maybe later we are going to create some extra features (but I think it is good to keep them in mind): 
  - install app from a autorun file => basically it is app installation
  - create a standalone app 
- XQuartz installation on OSX => I think we are not going to implement it but it means that we have to remove some wineversions 

If I see anything else, I'll post it here

**Comment by qparis** [comment]
Yes, and SetupWizard is a great tool for these task, especially for shortcut creation:

![image](https://user-images.githubusercontent.com/373645/50303578-cac02280-048d-11e9-9be8-2b64e1229caf.png)

**Comment by plata** [comment]
And you don't want setup wizard to appear in different contexts, right?

**Comment by qparis** [comment]
What do you mean by "in different context"?

**Comment by madoar** [comment]
I don't think we require a complete wizard with multiple steps for both the shortcut creation and the application removal.
I mean a wizard in my opinion can always be split into multiple parts that depend on the previous parts, which are shown in their own steps.
The issue here is, that this isn't the case for both the shortcut creation and the application removal.

**Comment by plata** [comment]
That the setup wizard is shown in a new window or the installations tab depending on what it does.

**Comment by qparis** [comment]
@plata The simplest, the better I guess..
